include/crpropa/module/Acceleration.h

```diff
@@ -0,0 +1,170 @@
+class DirectedFlowOfScatterCenters: public StepLengthModifier
+{
+	private:
+		Vector3d __scatterVelocity;
+	public:
+	DirectedFlowOfScatterCenters(const Vector3d &scatterCenterVelocity);
+	double modify(double steplength, Candidate* candidate);
+};
+
+
+/// @class QuasiLinearTheory
+/// @brief Scales the steplength according to quasi linear theory.
+/// @details Following quasi-linear theory [Schlickeiser1989], the mean free path \f$\lambda\f$ of a
+///  particle with energy \f$E\f$ and charge \f$Z\f$ in a field with turbulence spectrum
+///  \f$\frac{k}{k_{\min}}^{-q}\f$ is
+///   \f[ \lambda = {\left(\frac{B}{\delta B}\right)}^2 {\left(R_G\;  k_{\min}\right)}^{1-q} R_G \equiv \lambda_0 {\left( \frac{E}{1 EeV}\frac{1}{Z} \right)}^{2-q} \f]
+///  where \f$R_G = \frac{E}{B Z}\f$ is the gyro-radius of the
+///  particles.
+/// This class implements the rigidity dependent scaling factor used to modify
+/// the base step length.
+/// @par
+/// \b [Schlickeiser1989] R. Schlickeiser, Cosmic-Ray Transport and Acceleration. II.
+///  Cosmic Rays in Moving Cold Media with Application to Diffu-
+///  sive Shock Wave Acceleration, The Astrophysical Journal 336
+///  (1989) 264. doi:10.1086/167010.
+class QuasiLinearTheory : public StepLengthModifier
+{
+	private:
+		double __referenceEnergy;
+		double __turbulenceIndex;
+		double __minimumRigidity;
+
+	public:
+		QuasiLinearTheory(double referenecEnergy=1.*EeV, double turbulence_index=5./3, double minimumRigidity=0);
+		double modify(double steplength, Candidate* candidate);
+};
+
+
+
+
+/// @class ParticleSplitting
+/// @brief  Implements particle splitting, i.e. inverse thinning, to speed up
+//acceleration
+/// @details After crossing a surface a given number of times, the particle is
+/// split to N partilces with weight 1/N. This eases performance constraints in
+/// acceleration simulations due to the power law nature of many acceleration
+/// mechanisms.
+/// Thanks to Matthew Weiss, Penn State University for the first work on this
+/// feature in 2017.
+class ParticleSplitting: public Module
+{
+	int num_splits;
+	int crossing_threshold;
+	double min_weight;
+	ref_ptr<Surface> surface;
+	std::string counterid;
+
+	public:
+		/// @params surface							 The surface to monitor
+		/// @params crossing_threshold   Number of crossings after which a particle is split
+		/// @params num_splits					 Number of particles the candidate is split into
+		/// @params min_weight					 Minimum weight to consider. PArticles with
+		///															 a lower weight are not split again.
+		/// @params counterid						 An unique string to identify the particle
+		///															 property used for counting. Useful if
+		///                              multiple splitting modules are present.
+		ParticleSplitting(Surface *surface, int crossing_threshold=50,
+				int num_splits=5, double min_weight=0.01, std::string counterid="ParticleSplittingCounter");
```

Comment:
Same comment as before: if we can interact with a variable and use it as keyword in python, shouldn't we stick to our convention (camel case)?